src/main/java/com/example/hashcartapp/controller/AdvertisementController.java

```diff
@@ -0,0 +1,122 @@
+package com.example.hashcartapp.controller;
+
+import com.example.hashcartapp.dto.AdvertisementDTO;
+import com.example.hashcartapp.entities.Advertisement;
+import com.example.hashcartapp.repository.AdvertisementRepository;
+import com.example.hashcartapp.service.AdvertisementService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.HttpClientErrorException;
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import java.util.List;
+
+@RestController
+public class AdvertisementController {
+
+    @Autowired
+    private AdvertisementRepository advertisementRepository;
+
```

Comment:
if advertisementRepository bean is not used in this controller, please remove it. Always keep imports optmized and remove unused variables/beans